core/src/executor/evaluate/mod.rs

```diff
@@ -68,12 +68,22 @@ pub async fn evaluate<'a, T>(
             }
             .map(Evaluated::from)
         }
-        Expr::Subquery(query) => select(storage, query, context.as_ref().map(Rc::clone))
-            .await?
-            .map_ok(|row| row.take_first_value().map(Evaluated::from))
-            .next()
-            .await
-            .unwrap_or_else(|| Err(EvaluateError::NestedSelectRowNotFound.into()))?,
+        Expr::Subquery(query) => {
+            let evaluations = select(storage, query, context.as_ref().map(Rc::clone))
+                .await?
+                .map_ok(|row| row.take_first_value().map(Evaluated::from))
+                .take(2)
+                .try_collect::<Vec<_>>()
+                .await?;
+            match evaluations.len() {
+                0 => Err(EvaluateError::NestedSelectRowNotFound.into()),
+                1 => evaluations
+                    .into_iter()
+                    .nth(0)
+                    .unwrap_or_else(|| Err(EvaluateError::NestedSelectRowNotFound.into())),
```

Comment:
Thanks for suggetion of `take(2)` way.
But I'm not sure whether it's fine to use `NestedSelectRowNotFound` error again here🥲